### PR TITLE
The new and improved release 8.1 garbage collector

### PR DESCRIPTION
**Pull request by MichaelJCaruso**
This pull request releases the current, most reliable, version of the (8.1) garbage collector.  Additionally, it also includes a number of bug fixes and useful ancillary changes made in the process.  It builds on and largely completes the work of pull request #16 as well as integrating a number of changes already integrated into the release-8.0 branch of this repository in pull requests #18 and #20.

Among the significant garbage collector changes are the use of:
- only inter-container handle references during the _cdMark_ phase of the garbage collector to determine the subset of referenced handles not reachable from outside the container mesh (commit 5f089302b3b8ccf40a046ac7bd4b3a9874741af8).
- generation managed container marking and reclamation (commit  f2b62031319668ac8d82367379ab5f628062c2dd).

Bug fixes include the repair of:
- a pointer alignment problem in the frontend (e408a61f12505718253074c76f9f8004caf60a27 - present in the release-8.0 branch courtesy of pull request #18).
- an incorrect test in VContainerHandle::flushCachedResources (4ebea79525be4428c3e61912674a72619c8472c4 - present in the release-8.1 branch courtesy of pull request #16).
- an orphaned CPD in rtSELUV_Set::compact (42fea953a9f76e564d011130d26c06d4e4afa9b9 - present in the release-8.1 branch courtesy of pull request #16).
- an incorrectly NDFPathName deduced OSDPathName when the NDFPathName references the current directory (9c790458e487214ca43b7c8cda2ba58880e62a86 - present in the release-8.1 branch courtesy of pull request #16).

Ancillary changes include:
- better identification of platform specific system resource (particularly memory) utilization policies (7b866eb67c092f3ff0cba9f83da128f56fd3e10e).
- better and more comprehensive optional logging of garbage collector activity (4515e6441430141fa855bdf14f0ca3193a70332b - present in the release-8.1 branch courtesy of pull request #16).
- better update annotation modularity and availability (846aae3172845b67d30b1bba159eb066a10f2d09).
- better annotation of bootstrap updates (8d3fb57522518ab64e3a3cb2f59404a0535f4b4d - present in the release-8.0 branch courtesy of pull request #20).  
- the use of hard links in _pushBinaries_ when creating tool aliases (e.g., _onprof_ and _onck) (25ba45e3172e0b782baf790dcc27fbb57b5e7da9)

### CODE REVIEW

**Comment by VCommitter** [comment]
So bootstrapping a full investNet database is a more comprehensive test than what we did after we stopped erroneously counting CTE references as smart pointer references?

**Comment by MichaelJCaruso** [comment]
Prior to this fix, we couldn't even get that far.

**Comment by MichaelJCaruso** [comment]
Following up on the work done here, I've got another set of changes that use inter-handle references exclusively to compute estimated inter-handle reference counts in the garbage collector's cycle detect phase.  I can:

- Open a parallel pull request that contains everything in this pull request plus those changes.
- Add them now to this pull request.
- Wait until this request is accepted and then open a new request containing only those changes.

In all cases, those new changes are what we should be ultimately using since they do a more accurate job of estimating inter-handle reference counts; however, how they're accepted might make a difference in how we get to test them.

**Comment by VCommitter** [comment]
Can you provide a link to your inter-handle reference count changes? I'd like to study them before I make a recommendation for how to proceed.

**Comment by MichaelJCaruso** [comment]
They're in my release-8.1-gc branch (https://github.com/MichaelJCaruso/vision/tree/release-8.1-gc).  Compare them to my release-8.1 branch (https://github.com/MichaelJCaruso/vision/tree/release-8.1) which is also the source for pull request #19.

**Comment by VCommitter** [comment]
Using the latest GC changes the test `doOnceLeakCheck.vis` is failing again

**Comment by MichaelJCaruso** [comment]
Regarding the failure of the _doOnceLeakCheck.vis_ test, I've found the problem.  Basically, the set of container handles used to seed the handle reference count estimation phase of the garbage collector (Cycle Detect) wasn't queuing for examination some of the container handles needed to generate accurate handle reference count estimates.  Specifically, _EnqueuePossibleCycles_ was only queuing _*unreferenced*_ handles attached to unmarked _CTE_s when it should have been queuing _*all*_ handles attached to unmarked _CTE_s.  I'll submit the fix as soon as I've had a chance to do more testing, but I'm optimistic.

**Comment by VCommitter** [comment]
@MichaelJCaruso that's great news!

**Comment by VCommitter** [comment]
I just pulled from `MichaelJCaruso:release-8.1-gc` to do some testing of the latest changes that haven't yet made this pull request.  Very exciting stuff!  Hopefully you can take a look at pull request #22 and we can knock both of these out in the next couple days!!

**Comment by MichaelJCaruso** [comment]
Definitely will.

**Comment by MichaelJCaruso** [comment]
Just merged the memory leak fix (commit 47954ae58191d05b1fbb56f751ad141f79a7e61f) into this pull request's branch.  On to the next project :-)

**Comment by VCommitter** [comment]
All of my testing is very promising!  I ran all the backend tests that have been open sourced and many of those that have not.  Everything is passing and some memory usage spot checking is also promising.  My final idea for testing is to put some memory usage instrumentation into the IVR test suite (or maybe find the instrumentation that is already there because I recall that is how `doOnceLeakCheck.vis` came to be).

Once I get through more systematic memory testing I think the next step will to be to accept this pull request.  Unless some other reviewer is planning additional testing. 

Perhaps we should also tag the `release-8.1` branch with something like `8.1.0` before we merge this pull?  This is a fix, not an enhancement, but it is a significant fix.  Though if you think we should use some other strategy to manage bigger changes I'm open to suggestions.

**Comment by VCommitter** [comment]
_TL;DR: I believe we should accept this pull request after we tag the current code in `vision-dbms/vision:release-8.1`_ 

The broad efficiency testing is another feather in these changes cap. The testing has shown that, in aggregate, the improved garbage collector beats 8.1 without it for both performance (less than 1% improvement) and memory usage (about a 1% improvement).  It also tests within 1.5% of the 8.0 branch which doesn't have the extra layer of abstraction meant to facilitate object representation improvements.

When we originally introduced handle garbage collection it was because of a problem found doing this kind of memory and performance analysis so testing the changes this way gives me a lot of confidence the new handle garbage collection code doesn't reintroduce similar problems.

Here is a full dump of the test data I collected since I have access to a more extensive testdeck than what has been open sourced:

```
Stats for '8.0.0' calculated using 10 iterations
Stats for '8.1.0' calculated using 10 iterations
Stats for '8.1.1' calculated using 10 iterations
==================================================================================================================
                                                                     8.0.0               8.1.0               8.1.1
------------------------------------------------------------------------------------------------------------------
advanced gen-0             elapsedTime                               2.135               2.163               2.309
advanced gen-0             externalTime                              2.380               2.425               2.569
advanced gen-0             finishAllocation                 15,079,768.000      17,545,672.000      16,383,992.000
advanced gen-0             mappedAddressSpaceLevel          40,443,236.000      40,468,536.000      40,468,536.000
advanced gen-0             maxAllocation                    19,145,316.000      21,635,348.000      20,473,668.000
advanced gen-0             usageElapsedTime                          2.314               2.355               2.493
advanced gen-0             usageSystemTime                           0.088               0.084               0.098
advanced gen-0             usageUserTime                             1.944               1.976               2.116
aliasTest-0                elapsedTime                               1.378               1.354               1.428
aliasTest-0                externalTime                              1.624               1.586               1.697
aliasTest-0                finishAllocation                 14,318,492.000      16,423,548.000      15,417,776.000
aliasTest-0                mappedAddressSpaceLevel          37,876,524.000      37,894,268.000      37,894,268.000
aliasTest-0                maxAllocation                    18,155,264.000      20,276,420.000      19,270,648.000
aliasTest-0                usageElapsedTime                          1.558               1.520               1.624
aliasTest-0                usageSystemTime                           0.069               0.072               0.079
aliasTest-0                usageUserTime                             1.271               1.237               1.348
psx-20081210-0             elapsedTime                               5.177               5.184               5.514
psx-20081210-0             externalTime                              5.441               5.440               5.782
psx-20081210-0             finishAllocation                 21,245,464.000      24,555,192.000      22,929,640.000
psx-20081210-0             mappedAddressSpaceLevel          45,212,076.000      45,247,276.000      45,247,276.000
psx-20081210-0             maxAllocation                    27,762,592.000      30,968,820.000      29,360,092.000
psx-20081210-0             usageElapsedTime                          5.363               5.367               5.705
psx-20081210-0             usageSystemTime                           0.132               0.128               0.137
psx-20081210-0             usageUserTime                             4.770               4.826               5.126
constituentData-0          elapsedTime                               1.275               1.277               1.375
constituentData-0          externalTime                              1.519               1.513               1.638
constituentData-0          finishAllocation                 14,886,664.000      17,067,704.000      16,085,800.000
constituentData-0          mappedAddressSpaceLevel          37,963,400.000      37,982,920.000      37,982,920.000
constituentData-0          maxAllocation                    18,747,696.000      20,947,548.000      19,965,644.000
constituentData-0          usageElapsedTime                          1.452               1.444               1.566
constituentData-0          usageSystemTime                           0.070               0.073               0.079
constituentData-0          usageUserTime                             1.143               1.139               1.247
dataSetChanges-0           elapsedTime                               1.051               1.049               1.087
dataSetChanges-0           externalTime                              1.313               1.291               1.370
dataSetChanges-0           finishAllocation                 13,923,544.000      15,938,860.000      14,980,632.000
dataSetChanges-0           mappedAddressSpaceLevel          37,792,652.000      37,805,528.000      37,805,528.000
dataSetChanges-0           maxAllocation                    17,725,220.000      19,752,992.000      18,794,764.000
dataSetChanges-0           usageElapsedTime                          1.222               1.207               1.268
dataSetChanges-0           usageSystemTime                           0.058               0.061               0.062
dataSetChanges-0           usageUserTime                             0.946               0.950               1.017
defaultCurrency-0          elapsedTime                               1.435               1.472               1.587
defaultCurrency-0          externalTime                              1.683               1.713               1.850
defaultCurrency-0          finishAllocation                 14,661,332.000      16,842,104.000      15,841,812.000
defaultCurrency-0          mappedAddressSpaceLevel          41,593,408.000      41,600,680.000      41,600,680.000
defaultCurrency-0          maxAllocation                    18,342,792.000      20,514,708.000      19,519,376.000
defaultCurrency-0          usageElapsedTime                          1.615               1.643               1.779
defaultCurrency-0          usageSystemTime                           0.081               0.085               0.092
defaultCurrency-0          usageUserTime                             1.177               1.197               1.296
djnxxpq-20080820-0         elapsedTime                              14.903              14.977              15.284
djnxxpq-20080820-0         externalTime                             15.167              15.234              15.541
djnxxpq-20080820-0         finishAllocation                 55,387,352.000      60,662,316.000      58,150,364.000
djnxxpq-20080820-0         mappedAddressSpaceLevel          67,116,560.000      67,209,952.000      67,209,952.000
djnxxpq-20080820-0         maxAllocation                   118,925,784.000     119,597,296.000     117,088,112.000
djnxxpq-20080820-0         usageElapsedTime                         15.083              15.149              15.461
djnxxpq-20080820-0         usageSystemTime                           0.360               0.333               0.402
djnxxpq-20080820-0         usageUserTime                            13.611              14.066              14.547
emptyTest-0                elapsedTime                               0.210               0.217               0.216
emptyTest-0                externalTime                              0.460               0.472               0.476
emptyTest-0                finishAllocation                 10,213,128.000      10,290,128.000      10,093,152.000
emptyTest-0                mappedAddressSpaceLevel          35,880,852.000      35,880,852.000      35,880,852.000
emptyTest-0                maxAllocation                    13,120,764.000      13,197,776.000      13,000,800.000
emptyTest-0                usageElapsedTime                          0.395               0.398               0.401
emptyTest-0                usageSystemTime                           0.036               0.033               0.033
emptyTest-0                usageUserTime                             0.208               0.214               0.218
exch-20100804-0            elapsedTime                              11.662              12.103              12.108
exch-20100804-0            externalTime                             11.911              12.374              12.376
exch-20100804-0            finishAllocation                 30,033,648.000      35,533,100.000      32,846,616.000
exch-20100804-0            mappedAddressSpaceLevel          65,532,568.000      65,583,724.000      65,583,724.000
exch-20100804-0            maxAllocation                    52,571,316.000      57,929,896.000      55,278,552.000
exch-20100804-0            usageElapsedTime                         11.838              12.289              12.292
exch-20100804-0            usageSystemTime                           0.293               0.279               0.273
exch-20100804-0            usageUserTime                            10.770              11.051              11.212
exch-opn20110628-0         elapsedTime                               6.753               6.926               6.987
exch-opn20110628-0         externalTime                              7.009               7.181               7.253
exch-opn20110628-0         finishAllocation                 24,963,348.000      28,903,024.000      27,122,324.000
exch-opn20110628-0         mappedAddressSpaceLevel          49,995,748.000      50,053,128.000      50,053,128.000
exch-opn20110628-0         maxAllocation                    43,912,324.000      47,779,092.000      45,999,400.000
exch-opn20110628-0         usageElapsedTime                          6.939               7.107               7.171
exch-opn20110628-0         usageSystemTime                           0.176               0.174               0.174
exch-opn20110628-0         usageUserTime                             6.187               6.351               6.424
expirations-0              elapsedTime                               2.129               2.108               2.230
expirations-0              externalTime                              2.380               2.356               2.492
expirations-0              finishAllocation                 15,137,620.000      17,523,700.000      16,402,560.000
expirations-0              mappedAddressSpaceLevel          41,764,396.000      41,778,404.000      41,778,404.000
expirations-0              maxAllocation                    18,824,696.000      21,185,524.000      20,083,680.000
expirations-0              usageElapsedTime                          2.311               2.286               2.418
expirations-0              usageSystemTime                           0.088               0.082               0.094
expirations-0              usageUserTime                             1.833               1.847               1.961
fa-20110307-0              elapsedTime                              15.422              15.296              15.796
fa-20110307-0              externalTime                             15.692              15.601              16.102
fa-20110307-0              finishAllocation                152,465,580.000     155,414,840.000     154,081,744.000
fa-20110307-0              mappedAddressSpaceLevel         134,623,924.000     134,648,228.000     134,648,228.000
fa-20110307-0              maxAllocation                   406,079,288.000     404,885,792.000     403,552,696.000
fa-20110307-0              usageElapsedTime                         15.590              15.486              15.977
fa-20110307-0              usageSystemTime                           1.167               1.073               1.607
fa-20110307-0              usageUserTime                            13.166              13.473              13.562
tsfe-20091119-0            elapsedTime                              85.022              89.755              89.821
tsfe-20091119-0            externalTime                             85.514              90.096              90.191
tsfe-20091119-0            finishAllocation                201,722,720.000     205,756,964.000     202,995,500.000
tsfe-20091119-0            mappedAddressSpaceLevel         227,985,160.000     228,093,412.000     228,093,412.000
tsfe-20091119-0            maxAllocation                   524,286,388.000     524,128,760.000     521,368,780.000
tsfe-20091119-0            usageElapsedTime                         85.210              89.947              90.037
tsfe-20091119-0            usageSystemTime                           9.271              10.463               9.543
tsfe-20091119-0            usageUserTime                            74.232              77.401              79.419
hierarchyFeed-0            elapsedTime                               2.464               2.361               2.670
hierarchyFeed-0            externalTime                              2.714               2.609               2.939
hierarchyFeed-0            finishAllocation                 15,407,488.000      17,917,544.000      16,751,656.000
hierarchyFeed-0            mappedAddressSpaceLevel          41,925,952.000      41,945,896.000      41,945,896.000
hierarchyFeed-0            maxAllocation                    19,246,048.000      21,757,032.000      20,607,680.000
hierarchyFeed-0            usageElapsedTime                          2.644               2.540               2.863
hierarchyFeed-0            usageSystemTime                           0.096               0.089               0.100
hierarchyFeed-0            usageUserTime                             2.220               2.135               2.414
holdingsDeleteAndReload-0  elapsedTime                               2.901               3.061               3.174
holdingsDeleteAndReload-0  externalTime                              3.138               3.317               3.454
holdingsDeleteAndReload-0  finishAllocation                 16,049,024.000      18,561,540.000      17,393,496.000
holdingsDeleteAndReload-0  mappedAddressSpaceLevel          42,048,244.000      42,062,624.000      42,062,624.000
holdingsDeleteAndReload-0  maxAllocation                    19,661,060.000      22,143,220.000      20,999,328.000
holdingsDeleteAndReload-0  usageElapsedTime                          3.057               3.226               3.356
holdingsDeleteAndReload-0  usageSystemTime                           0.092               0.099               0.103
holdingsDeleteAndReload-0  usageUserTime                             2.593               2.708               2.843
membershipDeletion-0       elapsedTime                               2.324               2.375               2.542
membershipDeletion-0       externalTime                              2.573               2.627               2.811
membershipDeletion-0       finishAllocation                 15,237,676.000      17,701,400.000      16,486,564.000
membershipDeletion-0       mappedAddressSpaceLevel          38,171,444.000      38,198,368.000      38,198,368.000
membershipDeletion-0       maxAllocation                    19,247,200.000      21,735,764.000      20,520,928.000
membershipDeletion-0       usageElapsedTime                          2.485               2.535               2.715
membershipDeletion-0       usageSystemTime                           0.075               0.078               0.083
membershipDeletion-0       usageUserTime                             2.192               2.248               2.424
membershipWithoutDates-0   elapsedTime                               1.527               1.614               1.597
membershipWithoutDates-0   externalTime                              1.766               1.886               1.848
membershipWithoutDates-0   finishAllocation                 14,702,576.000      16,897,624.000      15,880,000.000
membershipWithoutDates-0   mappedAddressSpaceLevel          40,600,120.000      40,619,340.000      40,619,340.000
membershipWithoutDates-0   maxAllocation                    18,492,512.000      20,682,600.000      19,668,476.000
membershipWithoutDates-0   usageElapsedTime                          1.683               1.788               1.757
membershipWithoutDates-0   usageSystemTime                           0.073               0.076               0.081
membershipWithoutDates-0   usageUserTime                             1.292               1.367               1.379
missingFile-0              elapsedTime                               1.015               1.042               1.102
missingFile-0              externalTime                              1.245               1.305               1.373
missingFile-0              finishAllocation                 14,034,996.000      16,085,284.000      15,124,344.000
missingFile-0              mappedAddressSpaceLevel          37,828,076.000      37,843,320.000      37,843,320.000
missingFile-0              maxAllocation                    17,837,440.000      19,903,480.000      18,942,540.000
missingFile-0              usageElapsedTime                          1.181               1.233               1.291
missingFile-0              usageSystemTime                           0.065               0.072               0.073
missingFile-0              usageUserTime                             0.882               0.934               0.991
mcis-20070828-0            elapsedTime                              31.356              31.348              30.735
mcis-20070828-0            externalTime                             31.648              31.679              31.031
mcis-20070828-0            finishAllocation                 87,505,052.000     104,218,032.000      90,215,336.000
mcis-20070828-0            mappedAddressSpaceLevel          91,491,120.000      91,639,572.000      91,639,572.000
mcis-20070828-0            maxAllocation                   144,453,156.000     160,925,656.000     146,939,292.000
mcis-20070828-0            usageElapsedTime                         31.556              31.527              30.931
mcis-20070828-0            usageSystemTime                           0.504               0.740               0.631
mcis-20070828-0            usageUserTime                            29.471              29.671              29.415
mcis-20070828-1            elapsedTime                               2.885               2.962               3.023
mcis-20070828-1            externalTime                              3.017               3.095               3.185
mcis-20070828-1            finishAllocation                 50,547,384.000      70,341,456.000      68,308,432.000
mcis-20070828-1            mappedAddressSpaceLevel         109,191,808.000     109,588,884.000     109,549,568.000
mcis-20070828-1            maxAllocation                    56,369,992.000      76,191,632.000      74,158,608.000
mcis-20070828-1            usageElapsedTime                          2.953               3.025               3.101
mcis-20070828-1            usageSystemTime                           0.231               0.243               0.250
mcis-20070828-1            usageUserTime                             1.429               1.438               1.473
mcis-20070828-2            elapsedTime                              24.319              25.162              25.238
mcis-20070828-2            externalTime                             24.450              25.296              25.370
mcis-20070828-2            finishAllocation                 49,293,032.000      51,553,204.000      50,318,720.000
mcis-20070828-2            mappedAddressSpaceLevel         140,081,280.000     140,460,144.000     140,420,828.000
mcis-20070828-2            maxAllocation                   170,301,888.000     172,031,460.000     170,874,552.000
mcis-20070828-2            usageElapsedTime                         24.384              25.221              25.300
mcis-20070828-2            usageSystemTime                           2.175               2.734               2.730
mcis-20070828-2            usageUserTime                            21.607              22.255              22.117
mcis-20070828-3            elapsedTime                              31.235              31.203              30.235
mcis-20070828-3            externalTime                             31.405              31.368              30.390
mcis-20070828-3            finishAllocation                157,785,460.000     157,729,628.000     157,418,748.000
mcis-20070828-3            mappedAddressSpaceLevel         254,892,712.000     255,140,916.000     255,101,600.000
mcis-20070828-3            maxAllocation                   489,705,892.000     486,432,384.000     486,121,504.000
mcis-20070828-3            usageElapsedTime                         31.299              31.269              30.301
mcis-20070828-3            usageSystemTime                           2.075               2.010               2.014
mcis-20070828-3            usageUserTime                            28.768              28.154              27.335
mcis-20070828-4            elapsedTime                               3.101               3.230               3.324
mcis-20070828-4            externalTime                              3.241               3.374               3.475
mcis-20070828-4            finishAllocation                 46,403,664.000      65,118,616.000      62,909,960.000
mcis-20070828-4            mappedAddressSpaceLevel         321,540,952.000     322,016,000.000     321,970,160.000
mcis-20070828-4            maxAllocation                    52,278,344.000      70,985,164.000      68,753,280.000
mcis-20070828-4            usageElapsedTime                          3.171               3.302               3.401
mcis-20070828-4            usageSystemTime                           0.414               0.444               0.435
mcis-20070828-4            usageUserTime                             1.443               1.502               1.602
mcis-20070828-5            elapsedTime                               2.274               2.201               2.266
mcis-20070828-5            externalTime                              2.387               2.328               2.402
mcis-20070828-5            finishAllocation                 23,552,880.000      24,393,604.000      24,019,296.000
mcis-20070828-5            mappedAddressSpaceLevel         229,042,076.000     229,138,948.000     229,093,108.000
mcis-20070828-5            maxAllocation                    67,515,392.000      68,483,128.000      68,096,960.000
mcis-20070828-5            usageElapsedTime                          2.331               2.263               2.335
mcis-20070828-5            usageSystemTime                           0.426               0.181               0.282
mcis-20070828-5            usageUserTime                             1.747               1.806               1.753
mcis-20070828correction-0  elapsedTime                             132.628             139.572             133.068
mcis-20070828correction-0  externalTime                            132.920             139.909             133.384
mcis-20070828correction-0  finishAllocation                203,037,988.000     238,412,944.000     218,519,712.000
mcis-20070828correction-0  mappedAddressSpaceLevel         456,531,028.000     456,993,068.000     456,957,788.000
mcis-20070828correction-0  maxAllocation                   589,999,676.000     621,201,828.000     601,366,524.000
mcis-20070828correction-0  usageElapsedTime                        132.782             139.755             133.235
mcis-20070828correction-0  usageSystemTime                           7.746               5.512               6.362
mcis-20070828correction-0  usageUserTime                           118.039             129.156             120.483
patrick-20070316-0         elapsedTime                               5.398               5.404               5.428
patrick-20070316-0         externalTime                              5.653               5.656               5.712
patrick-20070316-0         finishAllocation                 18,675,128.000      22,184,100.000      20,695,500.000
patrick-20070316-0         mappedAddressSpaceLevel          45,230,336.000      45,281,172.000      45,281,172.000
patrick-20070316-0         maxAllocation                    23,679,256.000      26,280,032.000      24,791,432.000
patrick-20070316-0         usageElapsedTime                          5.580               5.584               5.635
patrick-20070316-0         usageSystemTime                           0.146               0.133               0.145
patrick-20070316-0         usageUserTime                             5.035               4.990               5.074
patrick-20070316-1         elapsedTime                               2.693               2.683               2.766
patrick-20070316-1         externalTime                              2.814               2.811               2.890
patrick-20070316-1         finishAllocation                 44,536,504.000      62,449,388.000      60,490,284.000
patrick-20070316-1         mappedAddressSpaceLevel          58,529,692.000      58,756,944.000      58,723,640.000
patrick-20070316-1         maxAllocation                    49,072,548.000      67,036,996.000      65,077,892.000
patrick-20070316-1         usageElapsedTime                          2.755               2.749               2.831
patrick-20070316-1         usageSystemTime                           0.222               0.203               0.214
patrick-20070316-1         usageUserTime                             1.255               1.252               1.293
patrick-20070316-2         elapsedTime                              13.855              14.138              13.981
patrick-20070316-2         externalTime                             13.995              14.278              14.120
patrick-20070316-2         finishAllocation                 77,864,852.000      79,331,328.000      78,800,392.000
patrick-20070316-2         mappedAddressSpaceLevel          98,139,784.000      98,328,816.000      98,295,512.000
patrick-20070316-2         maxAllocation                   212,941,672.000     212,301,552.000     211,770,616.000
patrick-20070316-2         usageElapsedTime                         13.917              14.201              14.049
patrick-20070316-2         usageSystemTime                           0.582               0.711               0.501
patrick-20070316-2         usageUserTime                            13.101              13.139              13.181
patrick-20070316-3         elapsedTime                               2.632               2.873               2.909
patrick-20070316-3         externalTime                              2.758               3.003               3.041
patrick-20070316-3         finishAllocation                 40,598,096.000      57,326,188.000      55,215,060.000
patrick-20070316-3         mappedAddressSpaceLevel         121,066,672.000     121,349,100.000     121,281,384.000
patrick-20070316-3         maxAllocation                    46,030,572.000      62,719,984.000      60,643,772.000
patrick-20070316-3         usageElapsedTime                          2.697               2.939               2.978
patrick-20070316-3         usageSystemTime                           0.363               0.387               0.392
patrick-20070316-3         usageUserTime                             1.231               1.280               1.296
patrick-20070316-4         elapsedTime                               0.975               0.972               1.000
patrick-20070316-4         externalTime                              1.112               1.112               1.137
patrick-20070316-4         finishAllocation                 31,334,444.000      31,983,040.000      31,630,968.000
patrick-20070316-4         mappedAddressSpaceLevel          86,044,852.000      86,064,904.000      86,005,380.000
patrick-20070316-4         maxAllocation                    44,982,156.000      45,652,736.000      45,326,264.000
patrick-20070316-4         usageElapsedTime                          1.037               1.036               1.062
patrick-20070316-4         usageSystemTime                           0.068               0.070               0.072
patrick-20070316-4         usageUserTime                             0.857               0.839               0.861
patrick-20070316-5         elapsedTime                               1.902               2.025               2.041
patrick-20070316-5         externalTime                              2.025               2.154               2.173
patrick-20070316-5         finishAllocation                 39,561,792.000      56,097,404.000      54,043,748.000
patrick-20070316-5         mappedAddressSpaceLevel         106,465,424.000     107,218,652.000     107,083,148.000
patrick-20070316-5         maxAllocation                    44,493,124.000      61,033,976.000      58,970,788.000
patrick-20070316-5         usageElapsedTime                          1.967               2.087               2.110
patrick-20070316-5         usageSystemTime                           0.149               0.168               0.151
patrick-20070316-5         usageUserTime                             1.120               1.171               1.196
sande-20070621-0           elapsedTime                              12.594              12.693              13.467
sande-20070621-0           externalTime                             12.849              12.948              13.738
sande-20070621-0           finishAllocation                 33,893,740.000      38,208,808.000      36,192,968.000
sande-20070621-0           mappedAddressSpaceLevel          57,248,640.000      57,321,272.000      57,321,272.000
sande-20070621-0           maxAllocation                    68,592,040.000      72,850,948.000      70,836,404.000
sande-20070621-0           usageElapsedTime                         12.771              12.867              13.653
sande-20070621-0           usageSystemTime                           0.576               0.339               0.372
sande-20070621-0           usageUserTime                            11.582              11.907              12.458
scaffoldSande-0            elapsedTime                               1.050               1.025               1.063
scaffoldSande-0            externalTime                              1.302               1.276               1.322
scaffoldSande-0            finishAllocation                 13,942,424.000      15,953,420.000      14,995,192.000
scaffoldSande-0            mappedAddressSpaceLevel          37,773,860.000      37,786,736.000      37,786,736.000
scaffoldSande-0            maxAllocation                    17,729,772.000      19,753,224.000      18,794,996.000
scaffoldSande-0            usageElapsedTime                          1.230               1.207               1.248
scaffoldSande-0            usageSystemTime                           0.063               0.071               0.076
scaffoldSande-0            usageUserTime                             0.926               0.919               0.952
scaffoldTo gen-0           elapsedTime                               2.565               2.641               2.927
scaffoldTo gen-0           externalTime                              2.817               2.887               3.206
scaffoldTo gen-0           finishAllocation                 15,774,676.000      18,735,208.000      17,212,620.000
scaffoldTo gen-0           mappedAddressSpaceLevel          38,372,996.000      38,407,656.000      38,407,656.000
scaffoldTo gen-0           maxAllocation                    19,933,448.000      22,927,212.000      21,404,624.000
scaffoldTo gen-0           usageElapsedTime                          2.731               2.801               3.104
scaffoldTo gen-0           usageSystemTime                           0.074               0.079               0.093
scaffoldTo gen-0           usageUserTime                             2.439               2.516               2.743
simpleHoldings-0           elapsedTime                               1.474               1.472               1.707
simpleHoldings-0           externalTime                              1.727               1.707               1.983
simpleHoldings-0           finishAllocation                 15,220,404.000      17,454,784.000      16,403,192.000
simpleHoldings-0           mappedAddressSpaceLevel          38,060,292.000      38,080,136.000      38,080,136.000
simpleHoldings-0           maxAllocation                    19,165,600.000      21,418,116.000      20,366,524.000
simpleHoldings-0           usageElapsedTime                          1.640               1.627               1.893
simpleHoldings-0           usageSystemTime                           0.071               0.070               0.084
simpleHoldings-0           usageUserTime                             1.348               1.326               1.533
pixie-20080630-0           elapsedTime                               6.183               6.232               6.555
pixie-20080630-0           externalTime                              6.435               6.472               6.865
pixie-20080630-0           finishAllocation                 21,654,672.000      25,037,020.000      23,381,628.000
pixie-20080630-0           mappedAddressSpaceLevel          49,556,860.000      49,615,196.000      49,615,196.000
pixie-20080630-0           maxAllocation                    28,258,172.000      31,506,232.000      29,851,532.000
pixie-20080630-0           usageElapsedTime                          6.366               6.402               6.770
pixie-20080630-0           usageSystemTime                           0.158               0.142               0.147
pixie-20080630-0           usageUserTime                             5.626               5.641               6.048

```
